File: 117-self-dividing-numbers/self.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        def divisible(num):
            for val in str(num):
                if val == '0' or num % int(val) > 0:
                    return False
            return True
        
        nums = []
        for num in range(left, right + 1):
            if divisible(num):
                nums.append(num)
        return nums

File: 117-self-dividing-numbers/self.ts
function selfDividingNumbers(left: number, right: number): number[] {
  let nums = [];
  
  while (left <= right) {
    if (left % 10 !== 0) {
      if (left.toString().split('').every(num => (left % Number(num)) === 0)) {
        nums.push(left);
      }
    }
    left += 1;
  }
  
  return nums;
};